parallel.cabal

```
@@ -44,6 +44,7 @@ library
     build-depends:
         array      >= 0.3 && < 0.6,
         base       >= 4.3 && < 4.17,
+        bifunctor                 ,
```

This should have a bound and only be used for GHC < 8.2 (since `Bifoldable` is included in `base` since GHC 8.2 / base-4.10). The most liberal bounds should be fine, since we only use `bifoldr` (at least in my suggestion).
```suggestion
        bifunctors >= 0.1 && < 5.7,
```
[bifunctor](https://hackage.haskell.org/package/bifunctor) is a deprecated package that doesn't even export the stuff we're using.